tools/functional-tester/etcd-tester/key_stresser.go

```diff
@@ -202,6 +210,87 @@ func newStressPut(kvc pb.KVClient, keySuffixRange, keySize int) stressFunc {
 	}
 }
 
+func newStressTxn(kvc pb.KVClient, keyTxnSuffixRange, depth int) stressFunc {
+	keys := make([]string, keyTxnSuffixRange)
+	for i := range keys {
+		keys[i] = fmt.Sprintf("/k%03d", i)
+	}
+	for i, k := range keys {
+		_, err := kvc.Put(context.Background(), &pb.PutRequest{
+			Key:   []byte(k),
+			Value: []byte{byte(i)},
+		}, grpc.FailFast(false))
+		if err != nil {
+			plog.Fatalf("newStressTxn error (%v)", err)
+		}
+	}
+	return writeTxn(kvc, keys, depth)
+}
+
+func writeTxn(kvc pb.KVClient, keys []string, depth int) stressFunc {
+	return func(ctx context.Context) (error, int64) {
+		ks := make(map[string]struct{}, depth)
+		for len(ks) != depth {
+			ks[keys[rand.Intn(64)]] = struct{}{}
+		}
+		selected := make([]string, 0, depth)
+		for k := range ks {
+			selected = append(selected, k)
+		}
+		com, delOp, putOp := getTxnReqs(selected[0], "bar00")
+		txnReq := &pb.TxnRequest{
+			Compare: []*pb.Compare{com},
+			Success: []*pb.RequestOp{delOp},
+			Failure: []*pb.RequestOp{putOp},
+		}
+
+		// add nested txns if any
+		for i := 1; i < depth; i++ {
+			k, v := selected[i], fmt.Sprintf("bar%02d", i)
+			com, delOp, putOp = getTxnReqs(k, v)
+			nested := &pb.RequestOp{
+				Request: &pb.RequestOp_RequestTxn{
+					RequestTxn: &pb.TxnRequest{
+						Compare: []*pb.Compare{com},
+						Success: []*pb.RequestOp{delOp},
+						Failure: []*pb.RequestOp{putOp},
+					},
+				},
+			}
+			txnReq.Success = append(txnReq.Success, nested)
+			txnReq.Failure = append(txnReq.Failure, nested)
+		}
+
+		_, err := kvc.Txn(ctx, txnReq, grpc.FailFast(false))
+		return err, int64(depth)
+	}
+}
+
+func getTxnReqs(key, val string) (com *pb.Compare, delOp *pb.RequestOp, putOp *pb.RequestOp) {
+	com = &pb.Compare{
+		Key:         []byte(key),
+		Target:      pb.Compare_VERSION,
```

Comment:
is version key version?

If version of key > 0 => key exists and then delete the key?
if version of key == 0 => no key, so put key there?

Reply:
also in what case where version of key == 0?

from line 219 `kvc.Put(context.Background(), &pb.PutRequest{`, all the key are written etcd already.